### DISCUSSION TITLE
How to programatically get UploadTree ID for extracting copyrights via API ( /uploads/{id}/item/{itemId}/copyrights)

### DISCUSSION THREAD

**Discussion by dineshr93** [fossology/fossology#2728] [Q&A] [answered]
I am trying to extract copyrights via API ( /uploads/{id}/item/{itemId}/copyrights). 

![image](https://github.com/fossology/fossology/assets/4498415/18ce8fb0-149f-420a-8269-69da5edec527)
I got upload id **3500** and am not sure what is UploadTree ID .
![image](https://github.com/fossology/fossology/assets/4498415/7b43b7a9-55fa-4c06-b14f-a9c068da26ac)

I tried by setting the item no **18854514** visible in the url while clicking a package in fossology 
![image](https://github.com/fossology/fossology/assets/4498415/62816972-791e-4a2f-bf23-23e02cff7323)

it is not working 
![image](https://github.com/fossology/fossology/assets/4498415/1384527f-17f7-4258-a85c-8fedf468adb1)
```
{
	"code": 400,
	"message": "ID must be a positive integer, 3500\/item\/18854514\/copyrights passed!",
	"type": "ERROR"
}
```

**Comment by dineshr93** [comment]
update: This api is working as intended in   `Version: [4.4.0], Branch: [HEAD], Commit: [#903df0] 2024/01/15 09:27` above error occured in  `fossology instance [4.2.1.0], Branch: [HEAD], Commit: [#9a0a6e]`
![image](https://github.com/fossology/fossology/assets/4498415/7ac7133b-bcee-4f39-9a9e-4f84e18c383d)

But still I am not sure how to get the ItemID of particular package in a folder via other api?
@dvjsharma / @GMishx  please may I have your help on this?

**Comment by GMishx** [comment, marked as answer]
You can get the item id for the top most item using `/uploads/{id}/topitem` endpoint.
